Проектная работа №1. Alex32163216

Alex32163216 opened the pull request:
Калькулятор счёта.
Код написан в 4-х классах. Считывает количество людей (на которых нужно разделить счет), название товаров, стоимости товаров. и в конце выводит все введение товары их стоимости, общую сумму товаров и итог деления общей суммы товаров на количество человек.

Commit:
Add dev branch

Alex32163216 committed:
Проектная работа № 1

gusar-off commented on src/main/java/Rub.java:
```diff
@@ -0,0 +1,23 @@
+        float amountForPersons;
+        String templatePut = "Итого, с каждого человека по %.2f";
+        amountForPersons = totalAmount / amountPersons;
+        switch ((int)amountForPersons % 10){
```
⚠️ Немного некорректно рассчитывается окончание. Получается если цена будет 14 рублей, метод вернет рубля. Правильный алгоритм звучит так - Проверить лежит ли остаток от деления на 100 в интервале от 11 до 14 включительно, если да, то возвращаем рублей, если нет, то прогоняем с твоими проверками. То есть нужно учесть, что числа заканчивающиеся на 11-12-13-14 и заканчивающиеся на 1-2-3-4, будут иметь разные окончания

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,51 @@
+        System.out.println("Введите названия и стоимости товаров (стоимости должны быть в формате рубли,копейки).");
+
+        while(true){
+            System.out.println("Введите название товара:");
```
В тексте, выводимом пользователю, лучше явно указать, что при после "завершить",  ввод товаров прекратится

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,51 @@
+                        totalPriceOfProducts += priceOfProduct;
+                        priceOfItem = (int) priceOfProduct;
+                        trailingSlashes = "";
+                        for(int i = nameOfProduct.length(); i<= 30; i++)
```
Можно заменить на метод стандартной библиотеки - `java.lang.String#repeat`

Alex32163216 committed:
Проектная работа № 1
Скорректированы ошибки с окончанием слово рубль,
добавлена информация, что "После ввода команды "завершить", ввод товаров прекратится.".

Alex32163216 committed:
Проектная работа № 1
Скорректированы ошибки с окончанием слово рубль,
добавлена информация, что "После ввода команды "завершить", ввод товаров прекратится.".
